From b76488806845b0c09879928d6a0a5440e34aa7e2 Mon Sep 17 00:00:00 2001
From: Maarten de Vries <maarten@de-vri.es>
Date: Wed, 27 Jul 2016 11:54:13 +0200
Subject: [PATCH] Allow blacklisting packages for testing.

---
 travis.sh | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/travis.sh b/travis.sh
index 53fd644..b71aa2f 100755
--- a/travis.sh
+++ b/travis.sh
@@ -33,6 +33,7 @@ if ! [ "$IN_DOCKER" ]; then
         -e CI_PARENT_DIR \
         -e UPSTREAM_WORKSPACE \
         -e TRAVIS_BRANCH \
+        -e TEST_BLACKLIST \
         -v $(pwd):/root/$REPOSITORY_NAME moveit/moveit_docker:moveit-$ROS_DISTRO-ci \
         /bin/bash -c "cd /root/$REPOSITORY_NAME; source .moveit_ci/travis.sh;"
     return_value=$?
@@ -142,7 +143,8 @@ my_travis_wait 60 catkin build --no-status --summarize || exit 1
 travis_run source install/setup.bash;
 
 # Only run tests on the current repo's packages
-TEST_PKGS=$(catkin_topological_order $CI_SOURCE_PATH --only-names)
+echo "Test blacklist: $TEST_BLACKLIST"
+TEST_PKGS=$(catkin_topological_order src --only-names | grep -Fvxf <(echo "$TEST_BLACKLIST" | tr ' ;,' '\n'))
 if [ -n "$TEST_PKGS" ]; then
     TEST_PKGS="--no-deps $TEST_PKGS";
 fi
-- 
2.9.0

